src/app/map/map.component.ts

```diff
@@ -1,144 +1,145 @@
-    });
-    this.markers = [];
-  }
-
-  ngAfterViewInit() {
-    this.getCurrentLocation();
-  }
-
-  getCurrentLocation() {
-    navigator.geolocation.getCurrentPosition(
-      (position: GeolocationPosition) => {
-
-        const point: google.maps.LatLngLiteral = {
-          lat: position.coords.latitude,
-          lng: position.coords.longitude,
-        };
-
-        this.mapCenter = new google.maps.LatLng(point);
-        this.map.panTo(point);
-      },
-      (error) => {
-        console.log("Could Not Find User Location " + error);
-      },
-      { enableHighAccuracy: true }
-    );
-  }
-
-  getNearbyBar(radius: number, openNow: boolean, center: any) {
-
-    let latty = center.lat.toString();
-    let longy = center.lng.toString();
-
-    const happyUrl = "https://maps.googleapis.com/maps/api/place/nearbysearch/json?key=AIzaSyCbOt9xUH2lyaTEKjdDOVK0BJ1dROGVxPY&location=" + latty + "," + longy + "&radius=" + radius + "&type=bar";
-    if (openNow) {
-      const URL = happyUrl + "&opennow";
-      fetch(URL).then(data => {
-        return data.json()
-      }).then(jsonData => {
-        this.createMarkersFromJson(jsonData.results);
-        console.log(jsonData.results);
-      }).catch(error => {
-        console.log(error);
-      })
-    } else {
-      const URL = happyUrl;
-      fetch(URL).then(data => {
-        return data.json()
-      }).then(jsonData => {
-        this.createMarkersFromJson(jsonData.results);
-        console.log(jsonData.results);
-      }).catch(error => {
-        console.log(error);
-      })
-    }
-
-  }
-
-  createMarkersFromJson(results: SearchResponse[]) {
-    results.map(obj => {
-      const point: google.maps.LatLngLiteral = {
-        lat: obj.geometry.location.lat,
-        lng: obj.geometry.location.lng
-      };
-      const currentPoint =  new google.maps.LatLng(point);
-      this.markers.push(currentPoint);
-    });
-  }
-
-  resetMarkers(){
-    this.markers = [];
-  }
-
-}
-
-// const point: google.maps.LatLngLiteral = {
-//   lat: position.coords.latitude,
-//   lng: position.coords.longitude,
-// };
-
-// this.mapCenter = new google.maps.LatLng(point);
\ No newline at end of file
+ import { Component, OnInit, ViewChild } from '@angular/core';
+ import { FormGroup } from '@angular/forms';
+ import { GoogleMap } from '@angular/google-maps';
+ import { Subscription } from 'rxjs';
+ import { Preferences } from '../interfaces/preferences.interface';
+ import { SearchResponse } from '../interfaces/search.response.interface';
+ import { PreferencesFormComponent } from '../preferences-form/preferences-form.component';
+ import { PrefFormService } from '../services/pref-form.service';
+
+ @Component({
+   selector: 'app-map',
+   templateUrl: './map.component.html',
+   styleUrls: ['./map.component.scss']
+ })
+ export class MapComponent implements OnInit {
+   @ViewChild(GoogleMap, { static: false }) map: GoogleMap;
+
+   center: google.maps.LatLngLiteral;
```

Comment:
for these types, come up with a consistent way, and then add a comment on each grouping explaining the contents of that group